.circleci/build_image.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+set -xe
+
+if [[ $# -ne 2 ]]; then
+  echo "Usage: $0 DOCKERFILE IMAGE_NAME"
+  echo ".dockerfile is appended to DOCKERFILE."
+  echo "janitortechnology/ is prepended to IMAGE_NAME."
+  exit 2
+fi
+
+DIR=`dirname $1`
+DOCKERFILE=`basename $1`
+IMAGE_NAME=$2
+
+if [[ -z $CIRCLE_PULL_REQUEST ]]; then
+  # Log in to Docker Hub.
+  docker login -u $DOCKER_USER -p $DOCKER_PASS
+  # Decide whether Docker Hub images should be tagged ":branch-Y" or ":latest".
+  if [[ $CIRCLE_BRANCH != "master" ]]; then
+    DOCKERHUB_TAG="branch-$CIRCLE_BRANCH"
+  else
+    DOCKERHUB_TAG="latest"
+  fi
+else
+  DOCKERHUB_TAG="pr"
+fi
+
+# Pull published image for cached layers. (Allowed to fail)
+docker pull janitortechnology/$IMAGE_NAME:latest || true
+if [[ $CIRCLE_JOB != "ubuntu-dev" ]]; then
+  # Load the `ubuntu-dev` base image from previous step.
+  docker load -i ubuntu-dev.tar
```

Comment:
We should execute this earlier so we don't need to pull ubuntu-dev from the Hub.